ui/components/SourceDetail.tsx

```diff
@@ -48,16 +48,25 @@ function SourceDetail({ className, name, info, type }: Props) {
 
   const items = info(s);
 
+  const isNameRelevant = (expectedName) => {
+    return expectedName == name
+  }
+
+  const isRelevant = (expectedType, expectedName) => {
+    return (expectedType == s.type && isNameRelevant(expectedName))
+  }
+
   const relevantAutomations = _.filter(automations, (a) => {
     if (!s) {
       return false;
     }
 
-    if (a?.sourceRef?.kind == s.type && a?.sourceRef?.name == name) {
+    if (type == "HelmChart" && isNameRelevant(a?.helmChart?.name)) {
```

Comment:
What's the difference between this check and the following `isNameRelevant` chain? Or why do we need that extra check?

Reply:
This page is used to show details from Sources(helm, git) but also Charts, which in this context is like another kind of source. For helm repositories and git repositories, we need to check for the sourceRef info as a way to filter elements. But for Charts there is no sourceRef to compare against, that is why this different check. I mean, there is but the sourceRef is for the chart field not for the release itself.